packages/vue-renderer/src/renderer.js

```diff
@@ -486,29 +266,29 @@ export default class VueRenderer {
     // Log rendered url
     consola.debug(`Rendering url ${url}`)
 
-    // Add url to the context
-    context.url = url
+    // Add url to the renderContext
+    renderContext.url = url
 
-    const { req = {} } = context
+    const { req = {} } = renderContext
 
-    // context.spa
-    if (context.spa === undefined) {
-      // TODO: Remove reading from context.res in Nuxt3
-      context.spa = !this.SSR || req.spa || (context.res && context.res.spa)
+    // renderContext.spa
+    if (renderContext.spa === undefined) {
+      // TODO: Remove reading from renderContext.res in Nuxt3
+      renderContext.spa = !this.SSR || req.spa || (renderContext.res && renderContext.res.spa)
     }
 
-    // context.modern
-    if (context.modern === undefined) {
-      context.modern = req.modernMode && this.context.options.modern === 'server'
+    // renderContext.modern
+    if (renderContext.modern === undefined) {
+      renderContext.modern = req._modern || this.context.options.modern === 'client'
     }
 
-    // Call context hook
-    await this.context.nuxt.callHook('vue-renderer:context', context)
+    // Call renderContext hook
+    await this.context.nuxt.callHook('vue-renderer:renderContext', renderContext)
```

Comment:
Breaking change because of route renaming?
```suggestion
    await this.context.nuxt.callHook('vue-renderer:context', renderContext)
```

Reply:
Nice point

Reply:
Nice point